qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -55,6 +56,15 @@ class QuantumKernel:
     """
 
     @deprecate_arguments("0.5.0", {"user_parameters": "training_parameters"})
+    @deprecate_function(
+        "The QuantumKernel class has been superseded by the "
+        "qiskit_machine_learning.kernels.FidelityQuantumKernel and "
+        "qiskit_machine_learning.kernels.TrainableFidelityQuantumKernel classes. "
+        "This class will be deprecated in a future release and subsequently "
+        "removed after that.",
+        stacklevel=3,
+        category=PendingDeprecationWarning,
```

Comment:
This is failing in Neko, due to `category` parameter that was recently added. Its not failing elsewhere in CI as presumably we test using the latest Terra from main. **_However_** the requirements.txt still defines Terra as >= 0.20 whereas this code really requires >= 0.22. One of these PRs that is introducing the use of newer features should change this - the one that will get merged first would be the ideal I guess.

Reply:
Yes, I didn't add this requirement here because PR Qiskit/qiskit-machine-learning#437 already does it.

Reply:
Yes, I was wondering, but since there was no comment with respect to #437 nor did it have a Hold label I was not sure.

Reply:
I added a comment to the description.